### PR TITLE
John conroy/home bar chart

### LINKED ISSUE #450: Implement home page visualization

### PR DESCRIPTION
**Pull request by john-conroy**
Thanks again to @keller-mark to pointing me to the vitessce vega-lite implementation. There are a few differences from the design. Fixes #450 .

- Tick label angle. Can't set the tick label angle to -30 without labels disappearing. The angle is set so that all labels will be displayed when the number of assay types is at its maximum of 23.
- Tick symbol. Kept the default single line instead of the ^ used in the design.
- Grid/tick/axis color. Using the transparent color provided in the design caused issues with overlapping lines. There is probably a way around this, but I don't think it's worth the effort of figuring out now.
- Axis title locations. I opted to keep them in the default locations until we add the functionality for users to select different data to visualize.

BarChartUpdated

### CODE REVIEW

File: context/app/static/js/components/Home/BarChart/BarChart.jsx [hunk 1]
```diff
@@ -1,8 +1,12 @@
...
-import { Placeholder } from './style';
+import { Vega as ReactVega } from 'react-vega';
 
-function BarChart() {
-  return <Placeholder />;
+function BarChart(props) {
+  const { data, spec: partialSpec } = props;
+  const spec = { ...partialSpec, data: { name: 'table' } };
+
+  // eslint-disable-next-line prettier/prettier
```
Comment (mccalluc) [inline context/app/static/js/components/Home/BarChart/BarChart.jsx:8]: why? If it doesn't work for us, should we tweak the prettier config?

File: context/app/static/js/components/Home/BarChartWrapper/BarChartWrapper.jsx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+  { doc_count: 19, key: 'MALDI-IMS-pos' },
+  { doc_count: 7, key: 'PAS' },
+  { doc_count: 23, key: 'bulk-RNA' },
+  { doc_count: 4, key: 'SNAREseq' },
+  { doc_count: 29, key: 'TMT-LC-MS' },
+  { doc_count: 7, key: 'Targeted-Shotgun-LC-MS' },
+  { doc_count: 12, key: 'LC-MS-untargeted' },
+  { doc_count: 22, key: 'WGS' },
+  { doc_count: 16, key: 'scRNA-Seq-10x' },
+  { doc_count: 8, key: 'sciATACseq' },
```
Comment (mccalluc) [inline context/app/static/js/components/Home/BarChartWrapper/BarChartWrapper.jsx:24]:
```suggestion
  { doc_count: 50, key: 'FAKE-AF-bulk' },
  { doc_count: 50, key: 'FAKE-ATACseq-bulk' },
  { doc_count: 50, key: 'FAKE-MxIF' },
  { doc_count: 40, key: 'FAKE-CODEX' },
  { doc_count: 40, key: 'FAKE-IMC' },
  { doc_count: 40, key: 'FAKE-MALDI-IMS-neg' },
  { doc_count: 30, key: 'FAKE-MALDI-IMS-pos' },
  { doc_count: 30, key: 'FAKE-PAS' },
  { doc_count: 30, key: 'FAKE-bulk-RNA' },
  { doc_count: 20, key: 'FAKE-SNAREseq' },
  { doc_count: 20, key: 'FAKE-TMT-LC-MS' },
  { doc_count: 20, key: 'FAKE-Targeted-Shotgun-LC-MS' },
  { doc_count: 10, key: 'FAKE-LC-MS-untargeted' },
  { doc_count: 10, key: 'FAKE-WGS' },
  { doc_count: 10, key: 'FAKE-scRNA-Seq-10x' },
  { doc_count: 0, key: 'FAKE-sciATACseq' },
```

I just worry about someone at a higher level seeing this and missing that it is all fake data.

File: context/package.json [hunk 3]
```diff
@@ -7,13 +7,17 @@
...
     "prop-types": "^15.7.2",
     "react": "^16.12.0",
     "react-dom": "^16.12.0",
     "react-twitter-embed": "^3.0.3",
+    "react-vega": "^7.3.0",
     "searchkit": "^2.4.1-alpha.4",
     "styled-components": "^5.1.0",
     "typeface-inter": "^3.12.0",
+    "vega": "^5.13.0",
+    "vega-lite": "^4.13.1",
```
Comment (mccalluc) [inline context/package.json:20]: Do these need to be here, or could NPM handle the transitive dependencies? I don't see them being explicitly imported in the code.

Comment (john-conroy) [reply context/package.json:20]: They are peer dependencies of `react-vega`

File: context/app/static/js/components/Home/BarChart/BarChart.jsx (hunk 1 above)
Comment (john-conroy) [reply context/app/static/js/components/Home/BarChart/BarChart.jsx]: Fixed.

**Comment by keller-mark** [comment]
Also maybe worth reducing the height slightly. On 13-inch macbook with my browser taking up the full screen, the X-axis labels are below the fold

Screen Shot 2020-06-23 at 4 43 05 PM

**Comment by keller-mark** [comment]
Also, in case it helps here, vega plots can be made responsive by setting `"width": "container"` https://vega.github.io/vega-lite/examples/bar_size_responsive.html

**Comment by john-conroy** [comment]
Thanks @keller-mark . I'll adjust the aspect ratio. `"width": "container"` wasn't working when I tried it, but I may have done something wrong. The alignment and title/tooltips (or lack thereof) are taken from the design.

**Comment by keller-mark** [comment]
Ok. I think those things can be added as minor issues later if necessary as well.

**Comment by john-conroy** [comment]
We can add tooltips on hover for the columns.

**Comment by mccalluc** [comment]
John, ping me on Slack tomorrow morning when this is merged, and I'll make the release. It looks good -- Thanks!

**Pull request merged**